holidays/countries/germany.py

```diff
@@ -102,6 +119,1200 @@ def __init__(self, *args, **kwargs) -> None:
         InternationalHolidays.__init__(self)
         StaticHolidays.__init__(self, GermanyStaticHolidays)
         super().__init__(*args, **kwargs)
+        # Disable automatic expansion after initial population to avoid
+        # implicitly populating adjacent years when checking membership
+        # (keeps per-year population semantics for school holiday checks).
+        self.expand = False
+
+    def _add_school_holidays_hh(self, year):
```

Comment:
To add school holidays on a per subdiv basis, you can use `def _populate_subdiv_{subdiv code}_public_holidays(self):` but I would recommend moving these after each subdiv-specific public holidays assignment
```suggestion
    def _populate_subdiv_hh_school_holidays(self):
```